contracts/core/RebalancingSetToken.sol

```diff
@@ -532,16 +522,9 @@ contract RebalancingSetToken is
         require(rebalanceState != State.Rebalance, "BURN_PAUSED_DURING_REBALANCE");
 
         // Require user has tokens to burn
-        require(balances[_from] >= _quantity, "NOT_ENOUGH_TOKENS_TO_BURN");
-
-        // Update token balance of user
-        balances[_from] = balances[_from].sub(_quantity);
-
-        // Update total supply of Set Token
-        totalSupply_ = totalSupply_.sub(_quantity);
+        require(balanceOf(_from) >= _quantity, "NOT_ENOUGH_TOKENS_TO_BURN");
```

Comment:
Should we get rid of this check here then?

Reply:
Yes, we can do that :)